src/dbnode/client/write_state.go

```diff
@@ -89,7 +105,7 @@ func (w *writeState) close() {
 
 	w.nsID.Finalize()
 	w.tsID.Finalize()
-
+	w.hostSuccessList = nil
```

Comment:
the point of pooling objects is to avoid allocations. don't reset to nil, reset this using the following code:

```
var emptyID ident.ID
for i := range w.hostSuccessList {
  w.hostSuccessList[i] = emptyID
}
w.hostSuccessList = w.hostSuccessList[:0]
```